src/main/java/io/cdap/plugin/gcp/common/GCPUtils.java

```diff
@@ -107,8 +107,10 @@ public static String extractPrivateKey(ServiceAccountCredentials credentials) {
     boolean isServiceAccountFilePath = SERVICE_ACCOUNT_TYPE_FILE_PATH.equals(serviceAccountType);
 
     for (String prefix : keyPrefix) {
-      if (isServiceAccountFilePath && serviceAccount != null) {
```

Comment:
`isServiceAccountFilePath` will be false when `serviceAccountType` is macro, but code is updated so when `serviceAccountType` is macro, it will not add any auth properties.

Reply:
So basically below check is exactly same as this one. Then why do we need that change?

Reply:
This change is required for the case when serviceAccountType is filePath and serviceAccount is null - for autodetect scenario. If we leave as it was it would fallback back to serviceAccountType as it was a JSON key.